app/src/main/java/org/connectbot/util/TerminalFontProvider.kt

```diff
@@ -0,0 +1,236 @@
+/*
+ * ConnectBot: simple, powerful, open-source SSH client for Android
+ * Copyright 2025 Kenny Root
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.connectbot.util
+
+import android.content.Context
+import android.graphics.Typeface
+import android.os.Handler
+import android.os.HandlerThread
+import android.util.Log
+import androidx.core.provider.FontRequest
+import androidx.core.provider.FontsContractCompat
+import org.connectbot.R
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Provider for loading terminal fonts via Google Fonts API.
+ * Fonts are cached after loading for performance.
+ * Supports both preset fonts (TerminalFont enum) and custom font names.
+ */
+class TerminalFontProvider(private val context: Context) {
```

Comment:
I'm suspicious that this class doesn't use `suspend fun` and also doesn't reference the `Dispatchers.IO` to touch the disk. Is this causing some jank in the UI? Currently the `ConsoleScreen.kt` is disabling `StrictMode` which might be hiding some problems in this code.